PSReadLine/ConsoleLib.cs

```diff
@@ -45,9 +45,15 @@ public virtual int CursorSize
                 else
                 {
                     _unixCursorSize = value;
-                    // Solid blinking block or blinking vertical bar
-                    Write(value > 50 ? "\x1b[2 q" : "\x1b[5 q");
                 }
+
+                // See the cursor ANSI codes at https://www.real-world-systems.com/docs/ANSIcode.html, searching for 'blinking block'.
+                // We write out the ANSI escape sequence even if we are on Windows, where the 'Console.CursorSize' API is supported by .NET.
+                // This is because this API works fine in console host, but not in Windows Terminal. The escape sequence will configure the
+                // cursor as expected in Windows Terminal, while in console host, the escape sequence works after it's written out, but then
+                // will be overwritten by 'CursorSize' when the user continues typing.
```

Comment:
Does this mean there's a bit of a race condition where it can look unexpectedly different before the user continues typing?

Reply:
There could be. The blinking underscore shape from Ansi code is thinner that "Console.CursorSize = 25" in console host, but not different too much.